apollo-router/src/plugins/rhai/engine.rs

```diff
@@ -1114,6 +1141,8 @@ impl Rhai {
             .register_global_module(module.into())
             // Register our base64 module (not global)
             .register_static_module("base64", base64_module.into())
+            // Register our expansion module (not global)
+            .register_static_module("expansion", expansion_module.into())
```

Comment:
I'm wondering if we should not just call it `environment` or  `env` and have something like `env.var("MY_VAR")`. `expansion` doesn't sound like something related to environment variable to me and also I think we're exposing internal details using `expansion`. Just my humble opinion here

Reply:
It's a fair point, but I don't want to expose it to `Rhai` as an `env` module because it actually isn't an `env` module. It's using the router's `expansion` module and that is giving us mediated access to `env`.

The motivation is to make it clear that you are getting behaviour which is different to what you would get from a un-mediated `env` implementation and that its behaviour is controlled by the way the router controls `env` expansion.

I'm not sure there's a right/wrong answer just different philosophies...

Reply:
Personally think this should be called `env`. The fact that it uses the expansion code under the hood is a detail that users won't really care about, and also `APOLLO_ROUTER_CONFIG_ENV_PREFIX` and `APOLLO_ROUTER_CONFIG_SUPPORTED_MODES` are undocumented unsupported variables currently.

Reply:
If the main objection is that the `expansion` module is mis-named, then perhaps we should consider renaming it? I think it's private to the crate, so a better name is open to us.

My guiding principle here is to try and maintain the name of the `router` modules that we expose it into `Rhai`. That will serve us well in the long term. Maintaining an unnecessary mapping of `router` module name to `Rhai` module name is going to be increasingly painful/confusing over time...

If, on the other hand, the idea is to present this function as a simple un-mediated `env` function, then that's what we should do and not make use of `expansion` at all.

We could even do both things, which might be the best answer.

To be clear: I'm opposed to presenting the `expansion` module functionality as an unmediated `env` function. It's misleading and likely to cause issues at some point in the future.

Reply:
I am fine with renaming the expansion module. It's a side effect of the original purpose, but this has now changed as the code is used not just for config expansion.
The users really won't have a clue what expansion means, so +1 to renaming aligning on env.

Reply:
Alternatively pull out the env stuff from the expansion module into a separate file and have `expansion` and `rhai` rely on it.

Reply:
Picking this up to try and get it over the line...

1. `var` is a reserved keyword in `Rhai`, so we can't use env.var. We could use `get`...
2. I think I could expose `expansion` as an `env` module, but I'd prefer to call it `router_env` to make it clear that it a mediated `env`. I'll also get rid of the `create` function and just create an expander every time the `get` function is called. Code would look like:
```
let value = router_env::get("whatever_env_you_want");
```
i.e.: it would be a `router_env` function, not a method.